### PR TITLE
Refactor Development section in Contributing guide

### LINKED ISSUE #2822: The contribution document doesn't indicate that how to run all checks
**Describe the bug**
The testing procedures in the CONTRIBUTING.md guide don't match the checks ran by CI system. 

**Steps to reproduce (if applicable)**
Changes in #2766 pass the checks suggested in the CONTRIBUTING.md guide but fail CI. 

**Expected behavior**
The CONTRIBUTING.md guide provides a set of steps that can help ensure that PR can successfully pass CI system.

### PR DESCRIPTION
**Pull request by imotov**
### Description

Removes UI commands form the Development section and adds a list of useful make commands.

### How was this PR tested?

By manually running commands added to the document.

Closes #2822

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,36 +15,47 @@ keep it open, just write `See #<Issue Number>`.
...
-1. To veryfy that all tests are passing run `make test-all` in the root directory.
-2. To fix code style and foramt as well as catch common mistakes run `make fix` in the root directory. Alternatively, run `make -k test-all docker-compose-down` to tear down the Docker services after running all the tests.
+## PR verification checks
+When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discoverd, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:
+1. To veryfy that all tests are passing run `make test-all`.
+2. To fix code style and foramt as well as catch common mistakes run `make fix`. Alternatively, run `make -k test-all docker-compose-down` to tear down the Docker services after running all the tests.
```
Comment (fulmicoton) [inline CONTRIBUTING.md:29]:
```suggestion
2. To fix code style and format as well as catch common mistakes run `make fix`. Alternatively, run `make -k test-all docker-compose-down` to tear down the Docker services after running all the tests.
```

**Review by fulmicoton — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,36 +15,47 @@ keep it open, just write `See #<Issue Number>`.
...
 [sign our CLA by visiting this link](https://cla-assistant.io/quickwit-oss/quickwit).
 
-# PR verification checks
-When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discoverd, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands:
-1. To veryfy that all tests are passing run `make test-all` in the root directory.
-2. To fix code style and foramt as well as catch common mistakes run `make fix` in the root directory. Alternatively, run `make -k test-all docker-compose-down` to tear down the Docker services after running all the tests.
+## PR verification checks
+When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discoverd, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:
+1. To veryfy that all tests are passing run `make test-all`.
```
Comment (fulmicoton) [inline CONTRIBUTING.md:28]:
```suggestion
1. To verify that all tests are passing run `make test-all`.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,36 +15,47 @@ keep it open, just write `See #<Issue Number>`.
...
 The process is simple and fast. Upon your first pull request, you will be prompted to
 [sign our CLA by visiting this link](https://cla-assistant.io/quickwit-oss/quickwit).
 
-# PR verification checks
-When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discoverd, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands:
-1. To veryfy that all tests are passing run `make test-all` in the root directory.
-2. To fix code style and foramt as well as catch common mistakes run `make fix` in the root directory. Alternatively, run `make -k test-all docker-compose-down` to tear down the Docker services after running all the tests.
+## PR verification checks
+When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discoverd, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:
```
Comment (fulmicoton) [inline CONTRIBUTING.md:27]:
```suggestion
When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discovered, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:
```

Comment (fulmicoton) [inline CONTRIBUTING.md:27]:
```suggestion
When you submit a pull request to the project, the CI system runs several verification checks. After your PR is merged, a more exhaustive list of test will be run.

You will be notified by email from the CI system if any issues are discovered, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:
```

Comment (imotov) [reply CONTRIBUTING.md]: I should have passed it through a spellchecker I guess 😄

**Pull request merged**